### PR TITLE
shell.nix update

### PR DESCRIPTION
**Pull request by fbegyn**
Updated the `shell.nix` file to be better. It is better, because it uses the mozilla nixpkgs overlay and installs `rls` and `rustfmt` along with `rustc` and `cargo`. Currently, the `rls` binary is missing from the nightly build of Rust, so inserted a workaround to be able to pin a specific nightly build.

Updated some documentation in relation to the `diesel` installation and added an example of how to use `.envrc` to ease setting up the dev environment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,3 +36,36 @@ The name is open for discussion.
    an admin user with password 'admin'.
 
 You can now start developing! A good way to start is to look at the routes defined in the [controllers](./src/controllers/).
+
+### Using `.envrc` for dev environment
```
Comment (rien) [inline README.md:40]: Is this `nix` only? If so, I would specify that so users that don't run nixos won't be confused.

Comment (fbegyn) [reply README.md:40]: [direnv](https://github.com/direnv/direnv) is a project that works on all OSes, so it's nothing Nixos specific in there AFAIK. It's based on the information up higher (to create the `zauth` user) and the [information](https://github.com/direnv/direnv/wiki/Postgresql) from `direnv` itself on how to use it for postgres.

The premise is that you can define the dev environment in a `.envrc` file and `direnv` will load and execute the content when entering the directory and unload everything when leaving the directory (it works with permissions, so when a file is changed you need to `direnv allow` it again before it does this, security and all).

Comment (rien) [reply README.md:40]: :+1:

**Review by rien — APPROVED** [review]

**Comment by fbegyn** [comment]
I might sneak the `rust-analyzer` and `rust-src` extensions in here as well, just to allow users the choice between using `rls` or `rust-analyzer`.

**Pull request merged**